src/pages/ChangePassword.ts

```diff
@@ -0,0 +1,154 @@
+import ProfileForm from '../components/form/profile-form';
+import Input from '../components/input';
+import List from '../components/list';
+import LineInput from '../components/lineinput';
+import Button from '../components/button';
+import {REG_EXP, ERROR_MSG, Field, FieldBlock} from '../utils/validationConst';
+import LeftNav from '../components/leftnav';
+import Avatar from '../components/avatar';
+import { PATH } from '../router/paths';
+import userApi from '../api/user-api';
+import Router from '../router';
+import { User } from '../core/types';
+import Form from '../components/form';
+
+export default class ChangePassword extends List {
+	router = new Router('#root');
+	constructor(props: {user?: User} = {}) {
+
+		const classAvatar = 'avatar';
+		const avatarProps = {};
+		const avatar: Avatar = new Avatar(
+			avatarProps,
+			'div',
+			classAvatar
+		);
+		
+		const fields: Field[] = [
+				{
+					type: 'password', 
+					label: 'Старый\u00A0пароль', 
+					name: 'oldPassword', 
+					value: '', 
+					valid: REG_EXP.PASSWORD, 
+					errorMsg: ERROR_MSG.PASSWORD, 
+					autocomplete: 'current-password'
+				},
+				{
+					type: 'password', 
+					label: 'Новый\u00A0пароль', 
+					name: 'newPassword', 
+					value: '', 
+					valid: REG_EXP.PASSWORD, 
```

Comment:
как насчет переименовать это поле? сейчас при чтении кода кажется, что в нем хранится либо boolean переменная, либо функция, возвращающая boolean переменную, хотя на самом деле там режит regexp. Предлагаю назвать ее так:
```suggestion
					validationRegexp: REG_EXP.PASSWORD, 
```

Reply:
Исправил